v2/component/launcher.go

```diff
@@ -251,24 +263,127 @@ func (l *Launcher) RunComponent(ctx context.Context, cmdArgs []string) error {
 	}
 	glog.Infof("ExecutorInput:\n%s\n=====", executorInput.String())
 	outputParametersTypeMap := make(map[string]string)
-	for outputParameterName, outputParameter := range l.runtimeInfo.OutputParameters {
-		outputParametersTypeMap[outputParameterName] = outputParameter.Type
+	for outputParamName, outputParam := range l.runtimeInfo.OutputParameters {
+		outputParametersTypeMap[outputParamName] = outputParam.Type
 
 	}
 	cacheKey, err := cacheutils.GenerateCacheKey(executorInput.GetInputs(), executorInput.GetOutputs(), outputParametersTypeMap, cmdArgs, l.options.ContainerImage)
 	if err != nil {
 		return fmt.Errorf("failure while generating CacheKey: %w", err)
 	}
-	_, err = cacheutils.GenerateFingerPrint(*cacheKey)
+	fingerPrint, err := cacheutils.GenerateFingerPrint(*cacheKey)
+	cachedMLMDExecutionID, err := l.cacheClient.GetExecutionCache(fingerPrint, l.options.PipelineName)
 	if err != nil {
-		return fmt.Errorf("failure while generating FingerPrint: %w", err)
+		return fmt.Errorf("failure while getting executionCache: %w", err)
+	}
+	if cachedMLMDExecutionID == "" {
+		return l.execute(ctx, executorInput, cmd, fingerPrint, args)
+	}
+
+	//To do, do we need to store pipeline task for such cached task
+	//To do, do we need to store MLMD artifacts, event types for such cached execution(I guess no)
```

Comment:
@Bobgy 
For steps that we found caches, 
1. Do we need to store MLMD output artifacts, output events types for such step?
2. Do we need to store pipeline task in BE for such step with caches?(if we don't store MLMD output artifacts for such step but we want to store such pipeline task in BE, we will to add more info in pipeline task to label this task is cached so that we won't fetch this task as a hit cache for future tasks, otherwise the cache logic won't work if we fetch task that is cached but there is no outputparamter/artifact stored in MLMD)

Reply:
1. We will
* create a new execution with state=cached
* create new events that connect the execution to existing artifacts

These make sure any downstream system (including UI)  do not need to distinguish cached/normal executions

2. It's up to you. It's simpler not to add it to task table, but we may want to add it in the future when we have use cases other than caching that needs this table.